DOCS-23222: Clarify use of `--inclusive` in `confluent iam rbac role-binding list`

stevebang opened the pull request:
Release Notes
-------------
Clarify the use of the `--inclusive` flag in the `confluent iam rbac role-binding list` CLI command. 
Fixes [DOCS-23222](https://confluentinc.atlassian.net/browse/DOCS-23222).

Also, updated all of the descriptions and examples. 

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----
<!--
Briefly describe **what** you have changed and **why**.
Optionally include your implementation strategy.
-->

References
----------
<!--
Copy and paste links to tickets, related PRs, GitHub issues, etc.
-->

Test & Review
-------------
<!--
Has it been tested? How?
Copy and paste any instructions or steps that can save the reviewer time.
-->

[DOCS-23222]: https://confluentinc.atlassian.net/browse/DOCS-23222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -88,28 +88,28 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
-	cmd.Flags().String("principal", "", "Principal whose role bindings should be listed.")
-	cmd.Flags().Bool("current-user", false, "Show role bindings belonging to the current user.")
-	cmd.Flags().String("role", "", "List role bindings under a specific role given to a principal. Or if no principal is specified, list principals with the role.")
+	cmd.Flags().String("principal", "", "Principal ID, which limits role bindings to this principal. If unspecified, list all principals and role bindings.")
```
The principal is not exactly an ID. It could look like `User:u-12345` which _contains_ an ID.

stevebang replied:
Updated -- that was pending a response in the #cli channel.

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -88,28 +88,28 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
-	cmd.Flags().String("role", "", "List role bindings under a specific role given to a principal. Or if no principal is specified, list principals with the role.")
+	cmd.Flags().String("principal", "", "Principal ID, which limits role bindings to this principal. If unspecified, list all principals and role bindings.")
+	cmd.Flags().Bool("current-user", false, "List role bindings assigned to the current user.")
+	cmd.Flags().String("role", "", "Predefined role assigned to `--principal`. If `--principal` is unspecified, list all principals assigned the role.")
```
There is a feature (really a bug, in my opinion) in the `pflag` library that doesn't allow us to use backticks in flag descriptions.
```suggestion
	cmd.Flags().String("role", "", `Predefined role assigned to "--principal". If "--principal" is unspecified, list all principals assigned the role.`)
```

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -88,28 +88,28 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
+		cmd.Flags().Bool("current-environment", false, "Use current environment ID for the environment scope.")
+		cmd.Flags().String("cloud-cluster", "", "Cloud cluster ID, which specifies the cloud cluster scope.")
+		cmd.Flags().String("kafka-cluster", "", "Kafka cluster ID, which specifies the Kafka cluster scope.")
+		cmd.Flags().String("schema-registry-cluster", "", "Schema Registry cluster ID, which specifies the SR cluster scope.")
```
```suggestion
		cmd.Flags().String("schema-registry-cluster", "", "Schema Registry cluster ID, which specifies the Schema Registry cluster scope.")
```

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -88,28 +88,28 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 
-	cmd.Flags().String("resource", "", "If specified with a role and no principals, list principals with role bindings to the role for this qualified resource.")
-	cmd.Flags().Bool("inclusive", false, "List all role bindings in a specific scope and its nested scopes.")
+	cmd.Flags().String("resource", "", "Resource type for . If specified with `--role` and no principals, list all principals and role bindings  the `--role` for the specified resource.")
```
Please revise... looks like some of the words got scrambled here.

stevebang replied:
Replaced $KAFKA_CLUSTER_ID with realistic Kafka cluster ID example

stevebang commented:
I checked other CLI commands and saw the use of the backtick and it displayed properly.

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -88,28 +88,28 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 
-	cmd.Flags().String("resource", "", "If specified with a role and no principals, list principals with role bindings to the role for this qualified resource.")
-	cmd.Flags().Bool("inclusive", false, "List all role bindings in a specific scope and its nested scopes.")
+	cmd.Flags().String("resource", "", "Resource type and identifier using "<Prefix>:<ID>" format (for example, "User:u-123456"). If specified with "--role" and no principals, list all principals and role bindings  the "--role" for the specified resource.")
```
```suggestion
	cmd.Flags().String("resource", "", "Resource type and identifier using "<Prefix>:<ID>" format (for example, "User:u-123456"). If specified with "--role" and no principals, list all principals and role bindings.")
```

stevebang commented:
@brianstrauch I'll add a DOCS ticket for me to add more/better examples.

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -31,7 +31,7 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 		Use:   "list",
 		Short: "List role bindings.",
-		Long:  "List the role bindings for a particular principal and/or role, and a particular scope.",
+		Long:  "List all role bindings assigned to a principal or role based on scopes.",
```
```suggestion
		Long:  "List role bindings assigned to a principal or role based on scopes.",
```

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -47,27 +47,27 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 			},
 			examples.Example{
-				Text: `List the role bindings for principals with role "CloudClusterAdmin":`,
+				Text: `List the role bindings for principals assigned the "CloudClusterAdmin" role:`,
```
```suggestion
				Text: `List the role bindings for principals with role "CloudClusterAdmin":`,
```

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -47,27 +47,27 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 			},
 			examples.Example{
-				Text: `List the role bindings for user "u-123456" with role "CloudClusterAdmin":`,
+				Text: `List the role bindings for user "u-123456" with the "CloudClusterAdmin" role:`,
```
```suggestion
				Text: `List the role bindings for user "u-123456" with role "CloudClusterAdmin":`,
```

brianstrauch commented:
Fix tests with `make integration-test INTEGRATION_TEST_ARGS="-run TestCLI/TestHelp -update"`

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -108,7 +108,7 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 	}
 
-	cmd.Flags().String("resource", "", "Resource type and identifier using "<Prefix>:<ID>" format (for example, "User:u-123456"). If specified with "--role" and no principals, list all principals and role bindings.")
+	cmd.Flags().String("resource", "", "Resource type and identifier using \"<Prefix>:<ID>\" format (for example, \"User:u-123456\"). If specified with "--role" and no principals, list all principals and role bindings.")
```
```suggestion
	cmd.Flags().String("resource", "", `Resource type and identifier using "<Prefix>:<ID>" format (for example, "User:u-123456"). If specified with "--role" and no principals, list all principals and role bindings.`)
```

stevebang replied:
In the docs, we have a mix of realistic and simplistic IDs. I'd rather have more realistic ones. I'm creating a list of IDs to use for examples. Do you have a list of the IDs used for examples in the CLI?

Also, I noted the use of single quotes to avoid the backslashes.

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/list-help-onprem.golden:
```diff
@@ -1,41 +1,41 @@
 List the role bindings for a specific principal:
 
-  $ confluent iam rbac role-binding list --kafka-cluster $KAFKA_CLUSTER_ID --principal User:my-user
+  $ confluent iam rbac role-binding list --kafka-cluster lkc-o3hw2k --principal User:my-user
```
This doesn't look right. These examples are for the on-prem version of the command, but we're using a Cloud resource ID here.

stevebang replied:
Changed to an example from the CP docs (`LRx92c9yQ+ws786HYosuBn`). Do you have a preferred generic example?

brianstrauch replied:
In a few of our examples, we just pad these IDs with 0s. There's no established format though...
```suggestion
  $ confluent iam rbac role-binding list --kafka-cluster 0000000000000000000000 --principal User:my-user
```

stevebang replied:
Changed to  0000000000000000000000. Not sure I like that, but better to be consistent for now.

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -59,57 +59,57 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 			examples.Example{
 				Text: "Only use the `--resource` flag when specifying a `--role` with no `--principal` specified. If specifying a `--principal`, then the `--resource` flag is ignored. To list role bindings for a specific role on an identified resource:",
-				Code: "confluent iam rbac role-binding list --kafka-cluster $KAFKA_CLUSTER_ID --role DeveloperRead --resource Topic",
+				Code: "confluent iam rbac role-binding list --kafka-cluster LRx92c9yQ+ws786HYosuBn --role DeveloperRead --resource Topic:my-topic",
```
```suggestion
				Code: "confluent iam rbac role-binding list --kafka-cluster 0000000000000000000000 --role DeveloperRead --resource Topic:my-topic",
```

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -59,57 +59,57 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 
-	cmd.Flags().String("resource", "", "If specified with a role and no principals, list principals with role bindings to the role for this qualified resource.")
-	cmd.Flags().Bool("inclusive", false, "List all role bindings in a specific scope and its nested scopes.")
+	cmd.Flags().String("resource", "", "Resource type and identifier using \"<Prefix>:<ID>\" format (for example, \"User:u-123456\"). If specified with \"--role\" and no principals, list all principals and role bindings.")
```
```suggestion
	cmd.Flags().String("resource", "", `Resource type and identifier using "<Prefix>:<ID>" format (for example, "User:u-123456"). If specified with "--role" and no principals, list all principals and role bindings.`)
```

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -59,57 +59,57 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 			examples.Example{
 				Text: "List the role bindings for a specific principal:",
-				Code: "confluent iam rbac role-binding list --kafka-cluster $KAFKA_CLUSTER_ID --principal User:my-user",
+				Code: "confluent iam rbac role-binding list --kafka-cluster LRx92c9yQ+ws786HYosuBn --principal User:my-user",
```
```suggestion
				Code: "confluent iam rbac role-binding list --kafka-cluster 0000000000000000000000 --principal User:my-user",
```

stevebang commented:
@brianstrauch I shortened descriptions for resource and principal (removed the examples) in the `list` command. Also, updated the resource and principal descriptions in the `create` and `delete` commands. Found another $KAFKA_CLUSTER_ID in the create example for CP and changed it to 0000000000000000000000.

sgagniere commented:
@stevebang @brianstrauch Do we want to merge this in for tomorrows release?

brianstrauch commented:
Up to @stevebang, looks like a few of the tests need fixing first though!

brianstrauch commented:
Any updates on this PR @stevebang?

brianstrauch commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
Please don't commit this file! (and feel free to add "/.vscode/" to the top of the .gitignore file to prevent this from happening again)

stevebang replied:
Sorry -- not sure why that appeared there, but I removed it.

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -59,57 +59,57 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 			examples.Example{
-				Text: "Only use the `--resource` flag when specifying a `--role` with no `--principal` specified. If specifying a `--principal`, then the `--resource` flag is ignored. To list role bindings for a specific role on an identified resource:",
-				Code: "confluent iam rbac role-binding list --kafka-cluster $KAFKA_CLUSTER_ID --role DeveloperRead --resource Topic",
+				Text: `Only use the "--resource" flag when specifying a "--role" with no "--principal" specified. If specifying a "--principal", then the "--resource" flag is ignored. To list role bindings for a specific role on an identified resource:`,
```
Please revert the changes on this line. We use code highlighting when flags and command names are wrapped in backticks, just like Slack

stevebang replied:
fixed

brianstrauch replied:
It doesn't look like these changes were added

stevebang replied:
fixed

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -59,57 +59,57 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
-	cmd.Flags().String("role", "", "List role bindings under a specific role given to a principal. Or if no principal is specified, list principals with the role.")
+	cmd.Flags().String("principal", "", "Principal ID, which limits role bindings to this principal. If unspecified, list all principals and role bindings.")
+	cmd.Flags().Bool("current-user", false, "List role bindings assigned to the current user.")
+	cmd.Flags().String("role", "", `Predefined role assigned to "--principal". If "--principal" is unspecified, list all principals assigned the role.`)
```
```suggestion
	cmd.Flags().String("role", "", "Predefined role assigned to `--principal`. If `--principal` is unspecified, list all principals assigned the role.")
```

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -59,57 +59,57 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 
-	cmd.Flags().String("resource", "", "If specified with a role and no principals, list principals with role bindings to the role for this qualified resource.")
-	cmd.Flags().Bool("inclusive", false, "List all role bindings in a specific scope and its nested scopes.")
+	cmd.Flags().String("resource", "", `Resource type and identifier using "<Prefix>:<ID>" format. If specified with "--role" and no principals, list all principals and role bindings.`)
```
```suggestion
	cmd.Flags().String("resource", "", "Resource type and identifier using \"<Prefix>:<ID>\" format. If specified with `--role` and no principals, list all principals and role bindings.")
```

JiaqiGao commented:
Seems like @brianstrauch has some new comments. Please review before merge @stevebang! Approved from side based on content.

stevebang commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -31,7 +31,7 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
 		Use:   "list",
 		Short: "List role bindings.",
-		Long:  "List the role bindings for a particular principal and/or role, and a particular scope.",
+		Long:  "List role bindings assigned to a principal or role based on scopes.",
```
No idea, but I'll remove that.

brianstrauch commented on internal/iam/command_rbac_role_binding_list.go:
```diff
@@ -59,57 +59,57 @@ func (c *roleBindingCommand) newListCommand() *cobra.Command {
-	cmd.Flags().String("role", "", "List role bindings under a specific role given to a principal. Or if no principal is specified, list principals with the role.")
+	cmd.Flags().String("principal", "", "Principal ID, which limits role bindings to this principal. If unspecified, list all principals and role bindings.")
+	cmd.Flags().Bool("current-user", false, "List role bindings assigned to the current user.")
+	cmd.Flags().String("role", "", "Predefined role assigned to '--principal'. If '--principal' is unspecified, list all principals assigned the role.")
```
```suggestion
	cmd.Flags().String("role", "", "Predefined role assigned to `--principal`. If `--principal` is unspecified, list all principals assigned the role.")
```

brianstrauch commented on internal/iam/command_rbac_role_binding_create.go:
```diff
@@ -24,46 +24,46 @@ func (c *roleBindingCommand) newCreateCommand() *cobra.Command {
 		exs = append(exs,
 			examples.Example{
-				Text: `Grant the role "CloudClusterAdmin" to the principal "User:u-123456" in the environment "env-123456" for the cloud cluster "lkc-123456":`,
+				Text: "Grant the role `CloudClusterAdmin` to the principal `User:u-123456` in the environment `env-123456` for the cloud cluster `lkc-123456`:",
```
We use quotes for resources and backticks for code snippets.
```suggestion
				Text: `Grant the role "CloudClusterAdmin" to the principal "User:u-123456" in the environment "env-123456" for the cloud cluster "lkc-123456":`,
```

brianstrauch commented on internal/iam/acl_utils.go:
```diff
@@ -36,7 +36,7 @@ access to the provided operations on the Kafka cluster itself.`)
 	flgSet.String("topic", "", `Set the topic resource. With this option the ACL grants the provided
 operations on the topics that start with that prefix, depending on whether
-the --prefix option was also passed.`)
+the "--prefix" option was also passed.`)
```
Thank you!

brianstrauch commented on internal/iam/command_rbac_role_binding_create.go:
```diff
@@ -64,17 +64,17 @@ func (c *roleBindingCommand) newCreateCommand() *cobra.Command {
 
 	cmd.Flags().String("role", "", "Role name of the new role binding.")
-	cmd.Flags().String("principal", "", "Qualified principal name for the role binding.")
+	cmd.Flags().String("principal", "", `Principal type and identifier using Prefix:ID format.`)
```
```suggestion
	cmd.Flags().String("principal", "", `Principal type and identifier using "Prefix:ID" format.`)
```

brianstrauch commented on internal/iam/command_rbac_role_binding_create.go:
```diff
@@ -64,17 +64,17 @@ func (c *roleBindingCommand) newCreateCommand() *cobra.Command {
+	cmd.Flags().String("principal", "", `Principal type and identifier using Prefix:ID format.`)
 	addClusterFlags(cmd, c.cfg, c.CLICommand)
-	cmd.Flags().String("resource", "", "Qualified resource name for the role binding.")
+	cmd.Flags().String("resource", "", `Resource type and identifier using Prefix:ID format.`)
```
```suggestion
	cmd.Flags().String("resource", "", `Resource type and identifier using "Prefix:ID" format.`)
```

brianstrauch commented on internal/iam/command_rbac_role_binding_delete.go:
```diff
@@ -35,10 +35,10 @@ func (c *roleBindingCommand) newDeleteCommand() *cobra.Command {
 
 	cmd.Flags().String("role", "", "Role name of the existing role binding.")
-	cmd.Flags().String("principal", "", "Qualified principal name associated with the role binding.")
+	cmd.Flags().String("principal", "", "Principal type and identifier using Prefix:ID format.")
```
```suggestion
	cmd.Flags().String("principal", "", "Principal type and identifier using "Prefix:ID" format.")
```

brianstrauch commented on internal/iam/command_rbac_role_binding_delete.go:
```diff
@@ -35,10 +35,10 @@ func (c *roleBindingCommand) newDeleteCommand() *cobra.Command {
 	pcmd.AddForceFlag(cmd)
 	addClusterFlags(cmd, c.cfg, c.CLICommand)
-	cmd.Flags().String("resource", "", "Qualified resource name for the role binding.")
+	cmd.Flags().String("resource", "", "Resource type and identifier using Prefix:ID format.")
```
```suggestion
	cmd.Flags().String("resource", "", "Resource type and identifier using "Prefix:ID" format.")
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/create-help-onprem.golden:
```diff
@@ -6,18 +6,18 @@ Usage:
 Flags:
       --role string                      REQUIRED: Role name of the new role binding.
-      --principal string                 REQUIRED: Qualified principal name for the role binding.
+      --principal string                 REQUIRED: Principal type and identifier using Prefix:ID format.
```
```suggestion
      --principal string                 REQUIRED: Principal type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/create-help-onprem.golden:
```diff
@@ -6,18 +6,18 @@ Usage:
       --cluster-name string              Cluster name to uniquely identify the cluster for role binding listings.
       --context string                   CLI context name.
-      --resource string                  Qualified resource name for the role binding.
+      --resource string                  Resource type and identifier using Prefix:ID format.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/create-help.golden:
```diff
@@ -42,15 +42,15 @@ Grant the "FlinkDeveloper" role to principal "User:u-123456" in environment "env
 Flags:
       --role string                      REQUIRED: Role name of the new role binding.
-      --principal string                 REQUIRED: Qualified principal name for the role binding.
+      --principal string                 REQUIRED: Principal type and identifier using Prefix:ID format.
```
```suggestion
      --principal string                 REQUIRED: Principal type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/create-help.golden:
```diff
@@ -42,15 +42,15 @@ Grant the "FlinkDeveloper" role to principal "User:u-123456" in environment "env
       --schema-registry-cluster string   Schema Registry cluster ID for the role binding.
       --ksql-cluster string              ksqlDB cluster name for the role binding.
-      --resource string                  Qualified resource name for the role binding.
+      --resource string                  Resource type and identifier using Prefix:ID format.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/delete-help-onprem.golden:
```diff
@@ -5,15 +5,15 @@ Usage:
 Flags:
       --role string                      REQUIRED: Role name of the existing role binding.
-      --principal string                 REQUIRED: Qualified principal name associated with the role binding.
+      --principal string                 REQUIRED: Principal type and identifier using Prefix:ID format.
```
```suggestion
      --principal string                 REQUIRED: Principal type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/delete-help-onprem.golden:
```diff
@@ -5,15 +5,15 @@ Usage:
       --cluster-name string              Cluster name to uniquely identify the cluster for role binding listings.
       --context string                   CLI context name.
-      --resource string                  Qualified resource name for the role binding.
+      --resource string                  Resource type and identifier using Prefix:ID format.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/delete-help.golden:
```diff
@@ -10,7 +10,7 @@ Delete the role "ResourceOwner" for the resource "Topic:my-topic" on the Kafka c
 Flags:
       --role string                      REQUIRED: Role name of the existing role binding.
-      --principal string                 REQUIRED: Qualified principal name associated with the role binding.
+      --principal string                 REQUIRED: Principal type and identifier using Prefix:ID format.
```
```suggestion
      --principal string                 REQUIRED: Principal type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/delete-help.golden:
```diff
@@ -19,7 +19,7 @@ Flags:
       --schema-registry-cluster string   Schema Registry cluster ID for the role binding.
       --ksql-cluster string              ksqlDB cluster name for the role binding.
-      --resource string                  Qualified resource name for the role binding.
+      --resource string                  Resource type and identifier using Prefix:ID format.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/delete-missing-role-cloud.golden:
```diff
@@ -9,7 +9,7 @@ Delete the role "ResourceOwner" for the resource "Topic:my-topic" on the Kafka c
 Flags:
       --role string                      REQUIRED: Role name of the existing role binding.
-      --principal string                 REQUIRED: Qualified principal name associated with the role binding.
+      --principal string                 REQUIRED: Principal type and identifier using Prefix:ID format.
```
```suggestion
      --principal string                 REQUIRED: Principal type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/delete-missing-role-cloud.golden:
```diff
@@ -18,7 +18,7 @@ Flags:
       --schema-registry-cluster string   Schema Registry cluster ID for the role binding.
       --ksql-cluster string              ksqlDB cluster name for the role binding.
-      --resource string                  Qualified resource name for the role binding.
+      --resource string                  Resource type and identifier using Prefix:ID format.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/list-failure-help-cloud.golden:
```diff
@@ -23,22 +23,22 @@ List the role bindings for user "u-123456" for all scopes:
+      --kafka-cluster string             Kafka cluster ID, which specifies the Kafka cluster scope.
+      --schema-registry-cluster string   Schema Registry cluster ID, which specifies the Schema Registry cluster scope.
+      --ksql-cluster string              ksqlDB cluster name, which specifies the ksqlDB cluster scope.
+      --resource string                  Resource type and identifier using Prefix:ID format. If specified with "--role" and no principals, list all principals and role bindings.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format. If specified with "--role" and no principals, list all principals and role bindings.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/list-failure-help-onprem.golden:
```diff
@@ -5,36 +5,36 @@ Usage:
       --context string                   CLI context name.
-      --resource string                  If specified with a role and no principals, list principals with role bindings to the role for this qualified resource.
-      --inclusive                        List all role bindings in a specific scope and its nested scopes.
+      --resource string                  Resource type and identifier using Prefix:ID format. If specified with "--role" and no principals, list all principals and role bindings.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format. If specified with "--role" and no principals, list all principals and role bindings.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/list-help-onprem.golden:
```diff
@@ -1,41 +1,41 @@
       --context string                   CLI context name.
-      --resource string                  If specified with a role and no principals, list principals with role bindings to the role for this qualified resource.
-      --inclusive                        List all role bindings in a specific scope and its nested scopes.
+      --resource string                  Resource type and identifier using Prefix:ID format. If specified with "--role" and no principals, list all principals and role bindings.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format. If specified with "--role" and no principals, list all principals and role bindings.
```

brianstrauch commented on test/fixtures/output/iam/rbac/role-binding/list-help.golden:
```diff
@@ -24,22 +24,22 @@ List the role bindings for user "u-123456" for all scopes:
+      --kafka-cluster string             Kafka cluster ID, which specifies the Kafka cluster scope.
+      --schema-registry-cluster string   Schema Registry cluster ID, which specifies the Schema Registry cluster scope.
+      --ksql-cluster string              ksqlDB cluster name, which specifies the ksqlDB cluster scope.
+      --resource string                  Resource type and identifier using Prefix:ID format. If specified with "--role" and no principals, list all principals and role bindings.
```
```suggestion
      --resource string                  Resource type and identifier using "Prefix:ID" format. If specified with "--role" and no principals, list all principals and role bindings.
```
